fdbclient/FileBackupAgent.actor.cpp

```diff
@@ -575,54 +576,78 @@ struct EncryptedRangeFileWriter : public IRangeFileWriter {
 		wPtr = mutateString(buffer);
 	}
 
-	static void validateEncryptionHeader(Optional<Reference<BlobCipherKey>> headerCipherKey,
+	static void validateEncryptionHeader(Reference<BlobCipherKey> headerCipherKey,
 	                                     Reference<BlobCipherKey> textCipherKey,
-	                                     BlobCipherEncryptHeader& header) {
+	                                     Optional<BlobCipherDetails> headerCipherDetails,
+	                                     BlobCipherDetails textCipherDetails) {
 		// Validate encryption header 'cipherHeader' details
-		if (header.cipherHeaderDetails.isValid() &&
-		    (!headerCipherKey.present() || header.cipherHeaderDetails != headerCipherKey.get()->details())) {
+		if (headerCipherKey.isValid() && headerCipherDetails.present() &&
+		    !(headerCipherDetails.get().baseCipherId == headerCipherKey->getBaseCipherId() &&
+		      headerCipherDetails.get().encryptDomainId == headerCipherKey->getDomainId() &&
+		      headerCipherDetails.get().salt == headerCipherKey->getSalt())) {
 			TraceEvent(SevWarn, "EncryptionHeader_CipherHeaderMismatch")
```

Comment:
nit: Please avoid using `_` in the TraceEvent messages; those logs shall be filtered and not supported